examples/structured_2d_dgsem/elixir_euler_warm_bubble.jl

```diff
@@ -0,0 +1,124 @@
+using OrdinaryDiffEq
+using Trixi
+
+# Physical constants
+g::Float64 = 9.81            # gravity of earth
+c_p::Float64 = 1004.0        # heat capacity for constant pressure (dry air)
+c_v::Float64 = 717.0         # heat capacity for constant volume (dry air)
+gamma::Float64 = c_p / c_v   # heat capacity ratio (dry air)
+
+# Warm bubble test case from
+# Wicker, L. J., and Skamarock, W. C.
+# A time-splitting scheme for the elastic equations incorporating second-order Runge–Kutta
+# time differencing
+# [DOI: 10.1175/1520-0493(1998)126%3C1992:ATSSFT%3E2.0.CO;2](https://doi.org/10.1175/1520-0493(1998)126%3C1992:ATSSFT%3E2.0.CO;2)
+function initial_condition_warm_bubble(x, t, equations::CompressibleEulerEquations2D)
+    # center of perturbation
+    center_x = 10000.0
+    center_z = 2000.0
+    # radius of perturbation
+    radius = 2000.0
+    # distance of current x to center of perturbation
+    r = sqrt((x[1] - center_x)^2 + (x[2] - center_z)^2)
+
+    # perturbation in potential temperature
+    potential_temperature_ref = 300.0
+    potential_temperature_perturbation = 0.0
+    if r <= radius
+        potential_temperature_perturbation = 2 * cospi(0.5 * r / radius)^2
+    end
+    potential_temperature = potential_temperature_ref + potential_temperature_perturbation
+
+    # Exner pressure, solves hydrostatic equation for x[2]
+    exner = 1 - g / (c_p * potential_temperature) * x[2]
+
+    # pressure
+    p_0 = 100_000.0  # reference pressure
+    R = c_p - c_v    # gas constant (dry air)
+    p = p_0 * exner^(c_p / R)
+
+    # temperature
+    T = potential_temperature * exner
+
+    # density
+    rho = p / (R * T)
+
+    v1 = 20.0
+    v2 = 0.0
+    E = c_v * T + 0.5 * (v1^2 + v2^2)
+    return SVector(rho, rho * v1, rho * v2, rho * E)
+end
+
+@inline function source_terms_gravity(u, x, t,
+                                      equations::CompressibleEulerEquations2D)
+    rho, _, rho_v2, _ = u
+    return SVector(zero(eltype(u)), zero(eltype(u)), -g * rho, -g * rho_v2)
+end
+
+###############################################################################
+# semidiscretization of the compressible Euler equations
+
+equations = CompressibleEulerEquations2D(gamma)
+
+boundary_conditions = (x_neg = boundary_condition_periodic,
+                       x_pos = boundary_condition_periodic,
+                       y_neg = boundary_condition_slip_wall,
+                       y_pos = boundary_condition_slip_wall)
+
+polydeg = 3
+basis = LobattoLegendreBasis(polydeg)
+
+surface_flux = FluxLMARS(340.0)
```

Comment:
Similar here. I could not find any mention for the reference sound speed in the docstring reference to be 340 m/s. Also, looking into the warm bubble setup from Knuth, he sets the reference sound speed to be `360.0` [here](https://github.com/OsKnoth/Trixi.jl/blob/369a8876d2585ae3c66864c2934fe0d67957660c/src/equations/2d/compressible_euler_moist.jl#L39). Where did you get this value for the `speed_of_sound` for the warm bubble?

Reply:
You are right!

This speed of sound only enters the LMARS flux. To be honest, I just viewed it as numerical parameter. IIRC @efaulhaber did some experiments. 

We cannot use the true, spatially varying, speed of sound here, but of course we could compute an averaged value from the initial state.

Reply:
> We cannot use the true, spatially varying, speed of sound here

Out of curiosity: Why not? Can't you just compute the speed of sound inside the flux (possibly as an average of the left and right states)?

Reply:
Sure, that was imprecise! The current implementation assumes a constant value. It could probably be done.

Reply:
Could you maybe leave a comment here how you got the estimate `340`?

Reply:
It is just the value that was used before (in @efaulhaber 's thesis). And it is the ISA value at sea level.

Reply:
IIRC, Oswald Knoth explained to me that this value is not relevant, as long as it's reasonably close to the actual speed of sound in the example. In the baroclinic instability example, I wrote
https://github.com/trixi-framework/Trixi.jl/blob/994bb4b088fbd08cf59296b8c5c6b3d33b810536/examples/p4est_3d_dgsem/elixir_euler_baroclinic_instability.jl#L223-L225
Or was it that we need any lower or upper bound or so? Anyway, the value was not particularly important and any reasonable estimate works.